tests/test_azure.py

```diff
@@ -58,37 +62,173 @@ def test_parse_image_name(monkeypatch):
+            pass
+        res_mgmt_client.resources = lambda: None
+        res_mgmt_client.resources.list_by_resource_group = lambda *args, **kwargs: items
+        return res_mgmt_client
+
+    def mock_compute_mgmt_client(self):
+        def compute_mgmt_client():
+            pass
+        compute_mgmt_client.images = lambda: None
+        compute_mgmt_client.images.delete = lambda rg, name: deleted_images.append(name)
+        return compute_mgmt_client
+
+    monkeypatch.setattr(Provider, 'cfgGet', mock_cfgGet)
+    monkeypatch.setattr(Azure, 'check_credentials', lambda *args, **kwargs: True)
+    monkeypatch.setattr(Azure, 'resource_mgmt_client', mock_res_mgmt_client)
+    monkeypatch.setattr(Azure, 'compute_mgmt_client', mock_compute_mgmt_client)
 
+    az = Azure('fake')
+    keep_images = ['SLES15-SP2-Azure-HPC.x86_64-0.9.1-Build1.3.vhd']
+    az.cleanup_images_from_rg(keep_images)
+    assert deleted_images == ['SLES15-SP2-Azure-HPC.x86_64-0.9.0-Build1.43.vhd']
+
+
+def test_cleanup_disks_from_rg(monkeypatch):
+    deleted_disks = list()
+
+    items = [
+            MockImage('SLES15-SP2-Azure-HPC.x86_64-0.9.0-Build1.43.vhd'),
+            MockImage('SLES15-SP2-Azure-HPC.x86_64-0.9.1-Build1.3.vhd'),
+            MockImage('SLES15-SP2-Azure-HPC.x86_64-0.9.1-Build1.7.vhd'),
+            MockImage('YouWillNotGetMyBuildNumber'),
+            ]
+
+    def mock_res_mgmt_client(self):
+        def res_mgmt_client():
+            pass
+        res_mgmt_client.resources = lambda: None
+        res_mgmt_client.resources.list_by_resource_group = lambda *args, **kwargs: items
+        return res_mgmt_client
+
+    def mock_compute_mgmt_client(self):
+        class FakeDisk:
+            def __init__(self, rg, name):
+                self.managed_by = name == 'SLES15-SP2-Azure-HPC.x86_64-0.9.0-Build1.43.vhd'
+
+        def compute_mgmt_client():
+            pass
+        compute_mgmt_client.disks = lambda: None
+        compute_mgmt_client.disks.get = lambda rg, name: FakeDisk(rg, name)
+        compute_mgmt_client.disks.delete = lambda rg, name: deleted_disks.append(name)
+        return compute_mgmt_client
 
-def test_cleanup_sle_images_container_too_young(monkeypatch):
-    test_name = 'young'
+    monkeypatch.setattr(Provider, 'cfgGet', mock_cfgGet)
     monkeypatch.setattr(Azure, 'check_credentials', lambda *args, **kwargs: True)
-    monkeypatch.setattr(BlockBlobService, 'list_blobs', list_blobs_mock)
-    monkeypatch.setattr(BlockBlobService, 'delete_blob', delete_blob_mock)
+    monkeypatch.setattr(Azure, 'resource_mgmt_client', mock_res_mgmt_client)
+    monkeypatch.setattr(Azure, 'compute_mgmt_client', mock_compute_mgmt_client)
+
+    keep_images = ['SLES15-SP2-Azure-HPC.x86_64-0.9.1-Build1.3.vhd']
+    az = Azure('fake')
+    az.cleanup_disks_from_rg(keep_images)
+    assert deleted_disks == ['SLES15-SP2-Azure-HPC.x86_64-0.9.1-Build1.7.vhd']
+
+
+def test_get_keeping_image_names(monkeypatch):
+    class FakeBlockBlobService:
```

Comment:
you keep re-implementing it in every test , isn't it differs on by last_modified date ?

Reply:
Most of the time it's a bit different.